src/flint/test/test_all.py

```diff
@@ -2930,14 +2962,18 @@ def quick_poly():
 
         assert +quick_poly() \
             == quick_poly()
+
         assert -quick_poly() \
-            == mpoly({(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4})
+            == mpoly(
+                {(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4} if P is not flint.nmod_mpoly
+                else {k: ctx.modulus() + v for k, v in {(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4}.items()}
+        )
```

Comment:
I would generally simplify these large expressions and bring out the conditional part to top level like:
```python
d = {(0, 0): -1, (0, 1): -2, (1, 0): -3, (2, 2): -4}
if P is not flint.nmod_mpoly:
    assert -quick_poly() == mpoly(d)
else:
    assert -quick_poly() == mpoly({m: v + ctx.modulus() for m, v in d.items()})
```
Does the `nmod_mpoly` constructor not take care of this automatically though?

I just checked:
```python
In [8]: ctx = nmod_mpoly_ctx(2, Ordering.lex, ['x', 'y'], 11)

In [9]: ctx.from_dict({(1, 2): 3})
Out[9]: 3*x*y^2

In [10]: ctx.from_dict({(1, 2): -3})
---------------------------------------------------------------------------
OverflowError                             Traceback (most recent call last)
Cell In[10], line 1
----> 1 ctx.from_dict({(1, 2): -3})

File nmod_mpoly.pyx:224, in flint.types.nmod_mpoly.nmod_mpoly_ctx.from_dict()

OverflowError: can't convert negative value to ulong
```
That should be fixed to be like other nmod types:
```python
In [11]: nmod_poly([0, -3], 11)
Out[11]: 8*x

In [12]: nmod(-3, 11)
Out[12]: 8

In [13]: nmod_mat([[1, 2], [3, -3]], 11)
Out[13]: 
[1, 2]
[3, 8]
```